digitalocean/resource_digitalocean_monitor_alert.go

```diff
@@ -0,0 +1,323 @@
+		Entities:    expandEntities(d.Get("entities").(*schema.Set).List()),
+	}
+
+	alertCreateRequest.Alerts = expandAlerts(d.Get("alerts").([]interface{}))
+
+	log.Printf("[DEBUG] Alert Policy create configuration: %#v", alertCreateRequest)
+	alertPolicy, _, err := client.Monitoring.CreateAlertPolicy(context.Background(), alertCreateRequest)
+	if err != nil {
+		return diag.Errorf("Error creating Alert Policy: %s", err)
+	}
+
+	d.SetId(alertPolicy.UUID)
+	log.Printf("[INFO] Alert Policy created, ID: %s", d.Id())
+
+	return resourceDigitalOceanMonitorAlertRead(ctx, d, meta)
+}
+
+func expandAlerts(config []interface{}) godo.Alerts {
+	alertConfig := config[0].(map[string]interface{})
+	alerts := godo.Alerts{
+		Slack: expandSlack(alertConfig["slack"].([]interface{})),
+		Email: expandEmail(alertConfig["email"].([]interface{})),
+	}
+	return alerts
+}
+
+func flattenAlerts(alerts godo.Alerts) []interface{} {
+	return []interface{}{
+		map[string]interface{}{
+			"email": flattenEmail(alerts.Email),
+			"slack": flattenSlack(alerts.Slack),
+		},
+	}
+}
+
+func expandSlack(slackChannels []interface{}) []godo.SlackDetails {
+	if len(slackChannels) == 0 {
+		return nil
+	}
+
+	expandedSlackChannels := make([]godo.SlackDetails, 0, len(slackChannels))
+
+	for _, slackChannel := range slackChannels {
+		slack := slackChannel.(map[string]interface{})
+		n := godo.SlackDetails{
+			Channel: slack["channel"].(string),
+			URL:     slack["url"].(string),
+		}
+
+		expandedSlackChannels = append(expandedSlackChannels, n)
+	}
+
+	return expandedSlackChannels
+}
+
+func flattenSlack(slackChannels []godo.SlackDetails) []interface{} {
+	result := make([]map[string]interface{}, 0, len(slackChannels))
+
+	for _, slackChannel := range slackChannels {
+		item := make(map[string]interface{})
+		item["url"] = slackChannel.URL
+		item["channel"] = slackChannel.Channel
+		result = append(result, item)
+	}
+
+	return []interface{}{result}
+}
+
+func expandEmail(config []interface{}) []string {
+	if len(config) == 0 {
+		return nil
+	}
+	emailList := make([]string, len(config))
+
+	for i, v := range config {
+		emailList[i] = v.(string)
+	}
+
+	return emailList
+}
+
+func flattenEmail(emails []string) *schema.Set {
+	if len(emails) == 0 {
+		return nil
+	}
+
+	flattenedEmails := schema.NewSet(HashStringIgnoreCase, []interface{}{})
+	for _, v := range emails {
+		if v != "" {
+			flattenedEmails.Add(v)
+		}
+	}
+
+	return flattenedEmails
+}
+
+func expandEntities(config []interface{}) []string {
+	alertEntities := make([]string, len(config))
+
+	for i, v := range config {
+		alertEntities[i] = v.(string)
+	}
+
+	return alertEntities
+}
+
+func flattenEntities(entities []string) []interface{} {
+	// it seems there are many functions like this in different places in the code base.
+	// maybe a utility library would be better
+	if len(entities) == 0 {
+		return nil
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"entities": entities,
+		},
+	}
+}
```

Comment:
Since [Line#308](https://github.com/digitalocean/terraform-provider-digitalocean/pull/679/files?file-filters%5B%5D=.go#diff-4f9285880d8045a1340ae2e5142505485dd0ed8a9dfe85d930a8ce90c2b4529bR306) can be changed, this funciton can be removed.
```suggestion
```